packages/hyperion-autologging/src/AutoLogging.ts

```diff
@@ -132,31 +130,35 @@ export function init(options: InitOptions): boolean {
     ALFlowletPublisher.publish(options.flowletPublisher);
   }
 
-  if (options.uiEventPublisher) {
-    ALUIEventPublisher.publish({
-      ...sharedOptions,
-      ...options.uiEventPublisher
-    });
-  }
-
-  if (options.heartbeat) {
-    ALHeartbeat.start(options.heartbeat);
-  }
-
   if (options.surfaceMutationPublisher) {
     ALSurfaceMutationPublisher.publish({
       ...sharedOptions,
       ...options.surfaceMutationPublisher
     });
   }
 
-  if (options.elementValuePublisher) {
+  if (options.uiEventPublisher) {
+    ALUIEventPublisher.publish({
+      ...sharedOptions,
+      ...options.uiEventPublisher
+    });
+
+    /**
+     * The following will depend on the surface mutation events
+     * so we need to make sure it is initialized afterwards
+     */
     ALElementValuePublisher.publish({
       ...sharedOptions,
-      ...options.elementValuePublisher
+      ...options.uiEventPublisher,
+      surfaceChannel: options.surface.channel,
     });
   }
 
+  if (options.heartbeat) {
+    ALHeartbeat.start(options.heartbeat);
```

Comment:
Unrelated to this specific PR, but related to the reason you moved this: Should probably also include a comment here that heartbeat relies on UI events?